# Rankine problem

## The question

A power plant based on the Rankine cycle is under development to provide a net power output of 10 MW. 
Solar collectors are to be used to generate Refrigerant 22 vapor at 1.6MPa, $50^\circ\text{C}$, for expansion through the turbine. Cooling water is available at $20^\circ\text{C}$. 
Specify the preliminary design of the cycle and estimate the thermal efficiency and the refrigerant and cooling water flow rates in kg/h.

## Step 1: Configure PYroMat

Unfortunately PYroMat does not have R22 multiphase tables, so I have opted for R134a instead as the multiphase tables are available in PYroMat.

In [1]:
import pyromat as pm

In [2]:
pm.config

     config_file : ['C:\\Users\\a-vanniekerk\\Anaconda3\\lib\\site-pa...
  config_verbose : False
         dat_dir : ['C:\\Users\\a-vanniekerk\\Anaconda3\\lib\\site-pa...
 dat_exist_fatal : False
   dat_overwrite : True
   dat_recursive : True
     dat_verbose : False
           def_T : 298.15
           def_p : 1.01325
     install_dir : 'C:\\Users\\a-vanniekerk\\Anaconda3\\lib\\site-pac...
         reg_dir : ['C:\\Users\\a-vanniekerk\\Anaconda3\\lib\\site-pa...
 reg_exist_fatal : False
   reg_overwrite : True
     reg_verbose : False
     unit_energy : 'kJ'
      unit_force : 'N'
     unit_length : 'm'
       unit_mass : 'kg'
     unit_matter : 'kg'
      unit_molar : 'kmol'
   unit_pressure : 'bar'
unit_temperature : 'K'
       unit_time : 's'
     unit_volume : 'm3'
         version : '2.0.12'

In [3]:
pm.config["unit_pressure"]="kPa"

In [4]:
R134a = pm.get("mp.C2H2F4")

## Step 2: Cycle set-up

The following points are assigned for the different stages of the rankine cycle under consideration:

point 1: before pump\
point 2: before solar collectors\
point 3: before turbine\
point 4: before condenser

Condenser saturated liquid temperature is assumed at $T_1=25^\circ\text{C}$.

## Step 3: Calculations

In [5]:
T1 = 25+273.15 #K
p1 = R134a.ps(T=T1)
v1 = 1/R134a.ds(T=T1)[0]
h1 = R134a.hs(T=T1)[0]
print(f"h1 = {round(float(h1),1)} kJ/kg")

h1 = 234.5 kJ/kg


In [6]:
p2 = 1.6*1000 #kPa
W_pump = v1*(p2-p1)
h2 = h1+W_pump
print(f"Work required by pump: {round(float(W_pump),1)} kJ/kg")

Work required by pump: 0.8 kJ/kg


In [7]:
p3 = p2
T3 = 50+273.15 #K
h3 = R134a.h(p=p3, T=T3)
s3 = R134a.s(p=p3, T=T3)
print(f"h3 = {round(float(h3),1)} kJ/kg")

h3 = 271.5 kJ/kg


In [8]:
q_H = h3-h2
print(f"Heat input by the solar collectors: {round(float(q_H),1)} kJ/kg")

Heat input by the solar collectors: 36.2 kJ/kg


In [9]:
s4 = s3
p4 = p1
x4 = R134a.T_s(p=p4, s=s4, quality=True)[1]
print(f"Quality of low pressure steam: {round(float(x4),4)}")

Quality of low pressure steam: 0.1952


In [10]:
h4 = R134a.h(p=p4, x=x4)
print(f"h4 = {round(float(h4),1)} kJ/kg")

h4 = 269.3 kJ/kg


In [11]:
W_turbine = h3-h4
print(f"Work generated by turbine: {round(float(W_turbine),1)} kJ/kg")

Work generated by turbine: 2.2 kJ/kg


In [12]:
eta_th = (W_turbine-W_pump)/q_H*100
print(f"Thermal efficiency is: {round(float(eta_th),2)}%")

Thermal efficiency is: 4.07%


## Reversible heat engine

We can test our thermal efficiency answer by using **Carnot's theorem**: 

> No engine operating between two heat reservoirs can be more efficient than a Carnot engine operating between the same reservoirs.

For a completely reversible heat engine the following is true: $$\eta_{th}=\frac{W}{Q_H}=1-\frac{T_C}{T_H}$$

In [13]:
T_H = T3
T_C = T1
eta_th = 1-T_C/T_H
print(f"Thermal efficiency is: {round(float(eta_th*100),2)}%")

Thermal efficiency is: 7.74%


The calculated efficiency of the rankine cycle of 4.07% is less than the maximum possible efficiency of 7.74%. The calculated efficiency is thus reasonable